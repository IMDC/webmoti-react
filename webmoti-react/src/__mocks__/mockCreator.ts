export function createMockParticipant(identity = 'mockIdentity', sid = 1, overrides = {}): any {
  return {
    identity,
    sid,
    tracks: new Map(),
    on: jest.fn(),
    off: jest.fn(),
    ...overrides,
  };
}

export function createMockRoom(name = 'mockRoom', localParticipant: any = null): any {
  return {
    name,
    localParticipant,
    participants: new Map(),
    on: jest.fn(),
    off: jest.fn(),
  };
}

export function createMockLocalTrack(
  kind: 'audio' | 'video',
  name = kind === 'video' ? 'camera' : 'microphone',
  overrides = {}
): any {
  return {
    kind,
    name,
    mediaStreamTrack: {
      label: `mock local ${kind} track`,
      getSettings: () => ({ deviceId: 'mock-device-id' }),
    },
    restart: jest.fn(),
    on: jest.fn(),
    off: jest.fn(),
    ...overrides,
  };
}

export function createMockConversation(): any {
  return {
    on: jest.fn(),
    off: jest.fn(),
  };
}
