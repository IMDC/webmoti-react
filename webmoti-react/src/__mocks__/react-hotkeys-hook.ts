export const useHotkeys = () => {
  // no-op mock since it's esm and makes jest fail
};
