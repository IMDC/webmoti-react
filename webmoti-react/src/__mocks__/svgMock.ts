const SvgMock = () => null;
export default SvgMock;
