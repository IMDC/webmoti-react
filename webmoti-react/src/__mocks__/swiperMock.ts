export const Pagination = jest.fn();
