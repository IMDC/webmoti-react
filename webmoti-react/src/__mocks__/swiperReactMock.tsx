export const Swiper = ({ children }: any) => <div>{children}</div>;
export const SwiperSlide = ({ children }: any) => <div>{children}</div>;
