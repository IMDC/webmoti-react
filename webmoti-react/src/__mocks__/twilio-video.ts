import { vi } from "vitest";
import { EventEmitter } from 'events';

class MockRoom extends EventEmitter {
  state = 'connected';
  disconnect = vi.fn();
  localParticipant = {
    publishTrack: vi.fn(),
    videoTracks: [{ setPriority: vi.fn() }],
  };
}

const mockRoom = new MockRoom();

class MockTrack extends EventEmitter {
  kind = '';
  stop = vi.fn();
  mediaStreamTrack = { getSettings: () => ({ deviceId: 'mockDeviceId' }) };

  constructor(kind: string) {
    super();
    this.kind = kind;
  }
}

const twilioVideo = {
  connect: vi.fn(() => Promise.resolve(mockRoom)),
  createLocalTracks: vi.fn(
    // Here we use setTimeout so we can control when this function resolves with jest.runAllTimers()
    () => new Promise(resolve => setTimeout(() => resolve([new MockTrack('video'), new MockTrack('audio')])))
  ),
  createLocalVideoTrack: vi.fn(() => new Promise(resolve => setTimeout(() => resolve(new MockTrack('video'))))),
};

export { mockRoom };
export default twilioVideo;
