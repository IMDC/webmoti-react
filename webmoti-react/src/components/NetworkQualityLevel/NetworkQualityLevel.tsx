import { styled } from '@mui/material/styles';
import { Participant } from 'twilio-video';

import useParticipantNetworkQualityLevel from '../../hooks/useParticipantNetworkQualityLevel/useParticipantNetworkQualityLevel';

const PREFIX = 'NetworkQualityLevel';

const classes = {
  outerContainer: `${PREFIX}-outerContainer`,
  innerContainer: `${PREFIX}-innerContainer`
};

const Root = styled('div')({
  [`&.${classes.outerContainer}`]: {
    width: '2em',
    height: '2em',
    padding: '0.9em',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    background: 'rgba(0, 0, 0, 0.5)',
  },
  [`& .${classes.innerContainer}`]: {
    display: 'flex',
    alignItems: 'flex-end',
    '& div': {
      width: '2px',
      marginRight: '1px',
      '&:not(:last-child)': {
        borderRight: 'none',
      },
    },
  },
});

const STEP = 3;
const BARS_ARRAY = [0, 1, 2, 3, 4];

export default function NetworkQualityLevel({ participant }: { participant: Participant }) {

  const networkQualityLevel = useParticipantNetworkQualityLevel(participant);

  if (networkQualityLevel === null) return null;

  return (
    <Root className={classes.outerContainer}>
      <div className={classes.innerContainer}>
        {BARS_ARRAY.map(level => (
          <div
            key={level}
            style={{
              height: `${STEP * (level + 1)}px`,
              background: networkQualityLevel > level ? 'white' : 'rgba(255, 255, 255, 0.2)',
            }}
          />
        ))}
      </div>
    </Root>
  );
}
