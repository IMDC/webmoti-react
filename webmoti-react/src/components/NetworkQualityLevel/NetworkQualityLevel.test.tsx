import { render } from '@testing-library/react';

import NetworkQualityLevel from './NetworkQualityLevel';
import { createMockParticipant } from '../../__mocks__/mockCreator';

describe('the NetworkQualityLevel component', () => {
  const mockParticipant = (level: number) =>
    createMockParticipant('mockparticipant', 1, { networkQualityLevel: level });

  it('should render correctly for level 5', () => {
    const { container } = render(<NetworkQualityLevel participant={mockParticipant(5)} />);
    expect(container).toMatchSnapshot();
  });

  it('should render correctly for level 3', () => {
    const { container } = render(<NetworkQualityLevel participant={mockParticipant(3)} />);
    expect(container).toMatchSnapshot();
  });

  it('should render correctly for level 0', () => {
    const { container } = render(<NetworkQualityLevel participant={mockParticipant(0)} />);
    expect(container).toMatchSnapshot();
  });
});
