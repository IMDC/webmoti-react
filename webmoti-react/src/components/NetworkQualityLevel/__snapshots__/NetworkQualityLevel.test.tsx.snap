// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the NetworkQualityLevel component should render correctly for level 0 1`] = `
<div>
  <div
    class="NetworkQualityLevel-outerContainer css-121vjg"
  >
    <div
      class="NetworkQualityLevel-innerContainer"
    >
      <div
        style="height: 3px; background: rgba(255, 255, 255, 0.2);"
      />
      <div
        style="height: 6px; background: rgba(255, 255, 255, 0.2);"
      />
      <div
        style="height: 9px; background: rgba(255, 255, 255, 0.2);"
      />
      <div
        style="height: 12px; background: rgba(255, 255, 255, 0.2);"
      />
      <div
        style="height: 15px; background: rgba(255, 255, 255, 0.2);"
      />
    </div>
  </div>
</div>
`;

exports[`the NetworkQualityLevel component should render correctly for level 3 1`] = `
<div>
  <div
    class="NetworkQualityLevel-outerContainer css-121vjg"
  >
    <div
      class="NetworkQualityLevel-innerContainer"
    >
      <div
        style="height: 3px; background: white;"
      />
      <div
        style="height: 6px; background: white;"
      />
      <div
        style="height: 9px; background: white;"
      />
      <div
        style="height: 12px; background: rgba(255, 255, 255, 0.2);"
      />
      <div
        style="height: 15px; background: rgba(255, 255, 255, 0.2);"
      />
    </div>
  </div>
</div>
`;

exports[`the NetworkQualityLevel component should render correctly for level 5 1`] = `
<div>
  <div
    class="NetworkQualityLevel-outerContainer css-121vjg"
  >
    <div
      class="NetworkQualityLevel-innerContainer"
    >
      <div
        style="height: 3px; background: white;"
      />
      <div
        style="height: 6px; background: white;"
      />
      <div
        style="height: 9px; background: white;"
      />
      <div
        style="height: 12px; background: white;"
      />
      <div
        style="height: 15px; background: white;"
      />
    </div>
  </div>
</div>
`;
